### PR TITLE
Refactor: replaces usages of Installable with Mod

### PR DESCRIPTION
**Pull request by mnholtz**
## What does this PR do?

- Part of #5934
- Replaces all usages of the word `Installable` (in type definitions, variable names, component names, etc.) with the word `Mod`
- Replaces some usages of the word `Blueprint` with the word `Mod` _only where_ `Blueprint` is used as a synonym for `Installable`

## Reviewer Tips

Changes to note/keep in mind:
- Watch the code walkthrough loom mentioned below
- Regarding the `extensionConsole/pages/blueprints` directory: most references to `blueprint` in filenames and component names are really a synonym for this "installable"/"mod" concept. The fact that it's called "blueprints page" is a legacy usage of the term "blueprint"
- I renamed the following redux slices:
    - `blueprintsSlice`
    - `blueprintModalsSlice`
    - Renaming blueprints slice is safe because it only saves the state of the Mods Page filtering/searching

## Code Walkthrough

https://www.loom.com/share/9154b1970b19430792d9d4c4192e188b

## Checklist

- [x] Add tests
- [x] Designate a primary reviewer @turbochef @grahamlangford

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -95,6 +95,11 @@ module.exports = {
       extends: ["pixiebrix/server"],
       rules: {
         "import/no-restricted-paths": "off",
+        "unicorn/prevent-abbreviations": {
+          allowList: {
+            mod: true,
+          },
+        },
```
Comment (mnholtz) [inline .eslintrc.js:102]: I added this in here quickly in order to get my ide to stop being upset about the word "mod". Not sure why the change to our eslint config isn't getting picked up

File: src/extensionConsole/pages/mods/ModsPageSidebar.tsx [hunk 2]
```diff
@@ -91,24 +91,22 @@ const ListItem: BsPrefixRefForwardingComponent<
...
     <FontAwesomeIcon className="align-self-center" icon={icon} />
     <span className="media-body ml-2">{label}</span>
   </Nav.Link>
 );
 
-const useOnboardingTabs = (
-  tableInstance: TableInstance<InstallableViewItem>
-) => {
+const useOnboardingTabs = (tableInstance: TableInstance<ModViewItem>) => {
   const { data: starterBlueprints, isLoading: isStarterBlueprintsLoading } =
```
Comment (grahamlangford) [inline src/extensionConsole/pages/mods/ModsPageSidebar.tsx:103]: Is there any reason we didn't change `starterBlueprints` to `starterMods` and `isStarterBlueprintsLoading` to `isStarterModsLoading`?

File: src/extensionConsole/pages/mods/ModsPageSidebar.tsx [hunk 3]
```diff
@@ -91,24 +91,22 @@ const ListItem: BsPrefixRefForwardingComponent<
...
   const [activeTab, setActiveTab] = useReduxState(
     selectActiveTab,
-    blueprintsSlice.actions.setActiveTab
+    modsSlice.actions.setActiveTab
   );
-  const { data: installableViewItems } = tableInstance;
+  const { data: modViewItems } = tableInstance;
 
   const {
     data: me,
```
Comment (grahamlangford) [inline src/extensionConsole/pages/mods/ModsPageSidebar.tsx:112]: We can also rename `onboardingBlueprintId` to `onboardingModId`

File: src/extensionConsole/pages/mods/hooks/useReactivateAction.ts [hunk 4]
```diff
@@ -15,32 +15,28 @@
...
-): (() => void | null) => {
+const useReactivateAction = (modViewItem: ModViewItem): (() => void | null) => {
   const dispatch = useDispatch();
   const { restrict } = useFlags();
-  const { installable, unavailable, status, sharing } = installableViewItem;
-  const isInstallableBlueprint = !isExtension(installable);
-  const hasBlueprint =
-    isExtensionFromRecipe(installable) || isInstallableBlueprint;
+  const { mod, unavailable, status, sharing } = modViewItem;
+  const hasBlueprint = isExtensionFromRecipe(mod) || isModDefinition(mod);
```
Comment (grahamlangford) [inline src/extensionConsole/pages/mods/hooks/useReactivateAction.ts:30]: `hasBlueprint` should be changed to `hasMod`

File: src/extensionConsole/pages/mods/hooks/useReactivateAction.ts [hunk 5]
```diff
@@ -15,32 +15,28 @@
...
     if (hasBlueprint) {
-      const blueprintId = isInstallableBlueprint
-        ? installable.metadata.id
-        : installable._recipe.id;
+      const blueprintId = isModDefinition(mod)
+        ? mod.metadata.id
+        : mod._recipe.id;
 
       reportEvent("StartInstallBlueprint", {
         blueprintId,
```
Comment (grahamlangford) [inline src/extensionConsole/pages/mods/hooks/useReactivateAction.ts:42]: Future work: creating an adapter used by reportEvent would allow us to pass `modId` instead.

File: src/extensionConsole/pages/mods/modsSelectors.ts
Comment (turbochef) [inline src/extensionConsole/pages/mods/modsSelectors.ts:1]: Very weird that github thought this .scss was renamed to this .ts file. It's not even close lol.

File: src/mods/useModViewItems.tsx [hunk 6]
```diff
@@ -150,12 +146,12 @@ function useInstallableViewItems(installables: Installable[]): {
   ]);
 
   return {
-    installableViewItems,
+    modViewItems: modViewItems,
```
Comment (turbochef) [inline src/mods/useModViewItems.tsx:149]:
```suggestion
    modViewItems,
```

File: src/utils/modUtils.test.ts [hunk 7]
```diff
@@ -15,27 +15,20 @@
...
 import { sharingDefinitionFactory } from "@/testUtils/factories/registryFactories";
 import { recipeDefinitionFactory } from "@/testUtils/factories/recipeFactories";
 
 describe("getSharingType", () => {
   test("personal extension", () => {
-    const installable: Installable = extensionFactory() as any;
+    const mod: Mod = extensionFactory() as any;
     const { type, label } = getSharingType({
-      installable,
+      mod: mod,
```
Comment (turbochef) [inline src/utils/modUtils.test.ts:31]:
```suggestion
      mod,
```

File: src/utils/modUtils.test.ts [hunk 8]
```diff
@@ -91,7 +84,7 @@ describe("getSharingType", () => {
     ];
 
     const { type, label } = getSharingType({
-      installable,
+      mod: mod,
```
Comment (turbochef) [inline src/utils/modUtils.test.ts:87]:
```suggestion
     mod,
```

File: src/utils/modUtils.test.ts [hunk 9]
```diff
@@ -116,7 +109,7 @@ describe("getSharingType", () => {
...
```
Comment (turbochef) [inline src/utils/modUtils.test.ts:112]:
```suggestion
      mod,
```

File: src/utils/modUtils.test.ts [hunk 10]
```diff
@@ -126,13 +119,13 @@ describe("getSharingType", () => {
...
-  test("public installable", () => {
-    const installable: Installable = recipeDefinitionFactory({
+  test("public mod", () => {
+    const mod: Mod = recipeDefinitionFactory({
       sharing: sharingDefinitionFactory({ public: true }),
     }) as any;
 
     const { type, label } = getSharingType({
-      installable,
+      mod: mod,
```
Comment (turbochef) [inline src/utils/modUtils.test.ts:128]:
```suggestion
     mod,
```

File: src/extensionConsole/pages/mods/ModsPageSidebar.tsx (hunk 2 above)
Comment (mnholtz) [reply src/extensionConsole/pages/mods/ModsPageSidebar.tsx:103]: Not a huge reason; I was hesitating because changing this would warrant also changing `starterBlueprints.ts` in the background script. Probably might as well include that

File: src/extensionConsole/pages/mods/GetStartedView.stories.tsx [hunk 11]
```diff
@@ -39,10 +39,7 @@ export default {
 function optionsStore(initialState?: any) {
   return configureStore({
     reducer: {
-      blueprints: persistReducer(
-        persistBlueprintsConfig,
-        blueprintsSlice.reducer
-      ),
+      mods: persistReducer(persistModsConfig, modsSlice.reducer),
```
Comment (grahamlangford) [inline src/extensionConsole/pages/mods/GetStartedView.stories.tsx:42]: Will this work with the persisted store?

File: src/extensionConsole/pages/mods/ModsPageSidebar.tsx (hunk 2 above)
Comment (mnholtz) [reply src/extensionConsole/pages/mods/ModsPageSidebar.tsx:103]: Actually, I'm going to leave this for future work. Changing the term "starterBlueprints" impacts consistency with messenger methods used by the app https://github.com/pixiebrix/pixiebrix-app/blob/5ebdb8f08fadb457f0bcb3f94d7dd9162ff2e845/pixiebrix-app/src/pages/onboarding/onboardingFlows.tsx#L199-L199

File: src/extensionConsole/pages/mods/hooks/useReactivateAction.ts (hunk 4 above)
Comment (mnholtz) [reply src/extensionConsole/pages/mods/hooks/useReactivateAction.ts:30]: In this case "blueprint" isn't exactly a synonym for "installable"... i'd sooner call this `hasModDefinition` or something?

File: .eslintrc.js (hunk 1 above)
Comment (mnholtz) [reply .eslintrc.js]: That was it 😆
